.github/workflows/unit-test-tools.yml

```diff
@@ -19,28 +19,48 @@ jobs:
         with:
           go-version: '^1.21.0'
 
-      - name: Build with TPG
+      - name: Build diff-processor with TPG
         run: |
           cd tools/diff-processor
           make clone OWNER_REPO=hashicorp/terraform-provider-google DEPTH=1
           make build OLD_REF=main NEW_REF=main
 
-      - name: Test with TPG
+      - name: Test diff-processor with TPG
         run: |
           cd tools/diff-processor
           go test -v ./...
         env:
           SERVICES_DIR: tools/diff-processor/new/google/services
 
-      - name: Build with TPGB
+      - name: Build diff-processor with TPGB
         run: |
           cd tools/diff-processor
           make clone OWNER_REPO=hashicorp/terraform-provider-google-beta DEPTH=1
           make build OLD_REF=main NEW_REF=main
 
-      - name: Test with TPGB
+      - name: Test diff-processor with TPGB
         run: |
           cd tools/diff-processor
           go test -v ./...
         env:
           SERVICES_DIR: tools/diff-processor/new/google/services
+
+      - name: Build go-changelog
```

Comment:
What would you think about doing these as separate jobs (instead of just separate steps?)

Reply:
Oh, like one GHA job per tool? Yeah, totally. I think I misunderstood the comment https://github.com/hashicorp/terraform-provider-google/issues/17598#issuecomment-2086555014 and thought we want to includes all unit test in tools directory into one job. I can split them up and also add one for `template-check`.

Reply:
I _think_ that comment was wondering if we could just run `go test` once and get the entire directory (so that we don't need to remember to add new tools to the unit test action in the future) but I'm not sure that would actually work.

Reply:
Unfortunately, it's not gonna work with running `go test` in the root. We'll need to have separate jobs or steps. Agree that having separate jobs may be clearer and more efficient for testing future tool changes in this case. I'll revert the changes for diff-processor and go-changlog but add new ones for [issue-labeler](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/tools/issue-labeler) and [template-check](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/tools/template-check) which are the two that contain unit tests but not running at this point.

Reply:
(To clarify, I was only suggesting new jobs within this file, not necessarily new files - although new files are necessary if we only want to run the tests for the specific modified tools.)

Reply:
oh, I see. That makes senses. Split them into separate jobs.